plugins/outputs/zabbix/utils.go

```diff
@@ -0,0 +1,230 @@
+	if name == "" {
+		return "", fmt.Errorf("empty measurement name")
+	}
+	if len(tags) == 0 {
+		return "", fmt.Errorf("metric without tags")
+	}
+
+	tagKeys := []string{}
+	for key, val := range tags {
+		if val != "" && key != "host" {
+			tagKeys = append(tagKeys, key)
+		}
+	}
+
+	if len(tagKeys) == 0 {
+		return "", fmt.Errorf("ignore metric with just 'host' tag")
+	}
+
+	sort.Strings(tagKeys)
+
+	lldKey := fmt.Sprintf("%s.%s", name, strings.Join(tagKeys, "."))
+
+	return lldKey, nil
+}
+
+// ElementsMatch compare list of elements ignoring order
+// Adapted from https://github.com/stretchr/testify/blob/85f2b59c4459e5bf57488796be8c3667cb8246d6/assert/assertions.go#L836
+func ElementsMatch(listA, listB interface{}) (ok bool) {
+	if isEmpty(listA) && isEmpty(listB) {
+		return true
+	}
+
+	aKind := reflect.TypeOf(listA).Kind()
+	bKind := reflect.TypeOf(listB).Kind()
+
+	if aKind != reflect.Array && aKind != reflect.Slice {
+		return false
+	}
+
+	if bKind != reflect.Array && bKind != reflect.Slice {
+		return false
+	}
+
+	aValue := reflect.ValueOf(listA)
+	bValue := reflect.ValueOf(listB)
+
+	aLen := aValue.Len()
+	bLen := bValue.Len()
+
+	if aLen != bLen {
+		return false
+	}
+
+	// Mark indexes in bValue that we already used
+	visited := make([]bool, bLen)
+	for i := 0; i < aLen; i++ {
+		element := aValue.Index(i).Interface()
+		found := false
+		for j := 0; j < bLen; j++ {
+			if visited[j] {
+				continue
+			}
+			if ObjectsAreEqual(bValue.Index(j).Interface(), element) {
+				visited[j] = true
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+
+	return true
+}
+
+// isEmpty gets whether the specified object is considered empty or not.
+func isEmpty(object interface{}) bool {
+	// get nil case out of the way
+	if object == nil {
+		return true
+	}
+
+	objValue := reflect.ValueOf(object)
+
+	switch objValue.Kind() {
+	// collection types are empty when they have no element
+	case reflect.Array, reflect.Chan, reflect.Map, reflect.Slice:
+		return objValue.Len() == 0
+		// pointers are empty if nil or if the value they point to is empty
+	case reflect.Ptr:
+		if objValue.IsNil() {
+			return true
+		}
+		deref := objValue.Elem().Interface()
+		return isEmpty(deref)
+		// for all other types, compare against the zero value
+	default:
+		zero := reflect.Zero(objValue.Type())
+		return reflect.DeepEqual(object, zero.Interface())
+	}
+}
+
+// ObjectsAreEqual determines if two objects are considered equal.
+// This function does no assertion of any kind.
+func ObjectsAreEqual(expected, actual interface{}) bool {
+	if expected == nil || actual == nil {
+		return expected == actual
+	}
+
+	exp, ok := expected.([]byte)
+	if !ok {
+		return reflect.DeepEqual(expected, actual)
+	}
+
+	act, ok := actual.([]byte)
+	if !ok {
+		return false
+	}
+	if exp == nil || act == nil {
+		return exp == nil && act == nil
+	}
+	return bytes.Equal(exp, act)
+}
```

Comment:
Wouldn't it be much easier to hash the []map[string]string elements you are actually comparing while you construct them? This reflection party is really not very efficient and also not necessary here as you _know_ the underlying type!

Reply:
Fixed in dd7f94be9